shared/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -499,6 +499,21 @@ namespace datadog::shared::nativeloader
         return S_OK;
     }
 
+    bool icompare_pred(WCHAR a, WCHAR b)
+    {
+        return std::tolower(a) == std::tolower(b);
+    }
+
+    bool string_iequal(WSTRING const& s1, WSTRING const& s2)
+    {
+        if (s1.length() == s2.length())
+        {
+            return std::equal(s2.begin(), s2.end(), s1.begin(), icompare_pred);
+        }
+
+        return false;
+    }
```

Comment:
I wonder if we could/should optimize this 🤔 e.g. we have a fixed set of constants we're embedding. If we're going to do a case insensitive compare, then we could

1. Store the constants as lowercase
2. Do a one-off toLower of the current process, and then do a simple case-sensitive compare.

Reply:
Okay I attempted to do that

For converting the constants to lowercase I went into VS Code, did a find `(["'])(.*?)\1` and selected all matches and then ran a `Transform to Lowercase` just to not do it manually incase of typos

I added a couple of new functions and I guess I can use them in the other project? It seemed to build fine but VS was yelling at me initially